### PR TITLE
Disable ext4 feature orphan_file for older releases

### LINKED ISSUE #278: Github Actions - Workflow test-build : 10 errors and 39 warnings
We get 10 errors and 39 warnings in the Github Workflow [test-build](https://github.com/grml/grml-debootstrap/actions/runs/9418427334):

The 10 errors come from:

```
test-debian (trixie, bullseye, mmdebstrap)
Process completed with exit code 1.
test-debian (trixie, stretch)
Process completed with exit code 1.
test-debian (trixie, buster)
Process completed with exit code 1.
test-debian (unstable, stretch)
Process completed with exit code 1.
test-debian (trixie, bullseye)
Process completed with exit code 1.
test-debian (unstable, bullseye, mmdebstrap)
Process completed with exit code 1.
test-debian (unstable, bullseye)
Process completed with exit code 1.
test-debian (trixie, buster, mmdebstrap)
Process completed with exit code 1.
test-debian (unstable, buster, mmdebstrap)
Process completed with exit code 1.
test-debian (unstable, buster) 
```

I took a closer look into the logs of `test-debian (unstable, buster)` and saw the following:
```
Run ./tests/build-vm-and-test.sh test
+ '[' '!' -d ./tests ']'
+ '[' test == setup ']'
+ RELEASE=buster
+ TARGET=qemu.img
+ DEBOOTSTRAP=
[...]
+ qemu-system-x86_64 -hda /home/runner/work/grml-debootstrap/grml-debootstrap/qemu.img -m 2048 -display none -vnc :0 -virtfs local,path=/tmp/tmp.viVk190zWP,mount_tag=host0,security_model=none,id=host0 -serial pty
+ echo 'No serial console from Qemu found yet [29 retries left]'
No serial console from Qemu found yet [29 retries left]
[...]
+ /home/runner/work/grml-debootstrap/grml-debootstrap/tests/serial-console-connection --tries 180 --port /dev/pts/0 --hostname buster --poweroff 'mount -t 9p -o trans=virtio,version=9p2000.L,rw host0 /mnt && cd /mnt && ./testrunner'
Login failure (try 0): Timeout exceeded.
[...]
Login failure (try 101): Timeout exceeded.
<pexpect.fdpexpect.fdspawn object at 0x7f7222758e50>
searcher: searcher_re:
    0: re.compile(b'buster login:')
Logging into /dev/pts/0 via serial console [try 0]
Waiting for login prompt...
Logging into /dev/pts/0 via serial console [try 1]
Waiting for login prompt...
Logging into /dev/pts/0 via serial console [try 2]
[...]
Logging into /dev/pts/0 via serial console [try 101]
Waiting for login prompt...
Login failure (try 102): Timeout exceeded.
<pexpect.fdpexpect.fdspawn object at 0x7f72227591b0>
searcher: searcher_re:
    0: re.compile(b'buster login:')
[...]
Logging into /dev/pts/0 via serial console [try 178]
Waiting for login prompt...
Logging into /dev/pts/0 via serial console [try 179]
Waiting for login prompt...
Error: Process completed with exit code 1.
```

I checked the logs of the successful run of `test-debian (unstable, bookworm)` for comparison:
```
Run ./tests/build-vm-and-test.sh test
+ '[' '!' -d ./tests ']'
+ '[' test == setup ']'
+ RELEASE=bookworm
+ TARGET=qemu.img
+ DEBOOTSTRAP=
[...]
+ qemu-system-x86_64 -hda /home/runner/work/grml-debootstrap/grml-debootstrap/qemu.img -m 2048 -display none -vnc :0 -virtfs local,path=/tmp/tmp.mnz6K2BXpF,mount_tag=host0,security_model=none,id=host0 -serial pty
+ '[' 30 -gt 0 ']'
+ (( timeout-- ))
+ grep -q 'char device redirected to ' qemu.log
+ echo 'No serial console from Qemu found yet [29 retries left]'
No serial console from Qemu found yet [29 retries left]
[...]
+ /home/runner/work/grml-debootstrap/grml-debootstrap/tests/serial-console-connection --tries 180 --port /dev/pts/0 --hostname bookworm --poweroff 'mount -t 9p -o trans=virtio,version=9p2000.L,rw host0 /mnt && cd /mnt && ./testrunner'
Login failure (try 0): Timeout exceeded.
<pexpect.fdpexpect.fdspawn object at 0x7fe70fab0520>
searcher: searcher_re:
    0: re.compile(b'bookworm login:')
Login failure (try 1): Timeout exceeded.
<pexpect.fdpexpect.fdspawn object at 0x7fe70fab0130>
searcher: searcher_re:
    0: re.compile(b'bookworm login:')
Login failure (try 2): Timeout exceeded.
<pexpect.fdpexpect.fdspawn object at 0x7fe70fab08b0>
searcher: searcher_re:
    0: re.compile(b'bookworm login:')
Logging into /dev/pts/0 via serial console [try 0]
Waiting for login prompt...
Logging into /dev/pts/0 via serial console [try 1]
Waiting for login prompt...
Logging into /dev/pts/0 via serial console [try 2]
Waiting for login prompt...
Logging into /dev/pts/0 via serial console [try 3]
Waiting for login prompt...
Logging in...
>> root
>> grml
Waiting for shell prompt...
>> 
<< b'\r\n'
<< b'\x1b[?2004l\r\x1b[?2004hroot@bookworm:~# '
Running commands...
[...]
+ EXIT_CODE=0
+ exit 0
```

It seems that there is a problem with the QEMU image (with the given HOST  / RELEASE / DEBOOTSTRAP combination).
I am not aware of any internals, but we might also hit some kind of (Github) qemu build limits (we spawn 34 workflows and QEMU instances).
@zeha Do you have an idea, what the problem might be?

The 39 warnings basically come from `actions/download-artifact@v3` + `actions/upload-artifact@v3`:

```
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/download-artifact@v3, actions/upload-artifact@v3. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.
```

So, we just need to migrate to [actions/download-artifact@v4](https://github.com/marketplace/actions/download-a-build-artifact) + [actions/upload-artifact@v4](https://github.com/marketplace/actions/upload-a-build-artifact).

I can try to test that migration and can come up with a PR if it works.

### PR DESCRIPTION
**Pull request by zeha**
Fixes #278.

### CODE REVIEW

**Comment by zeha** [comment]
Should be merged after #284

File: grml-debootstrap [hunk 1]
```diff
@@ -1301,36 +1301,49 @@ mkfs() {
...
-  if [ -n "$MKFS" ] && [ "$MKFS" = "mkfs.ext4" ] ; then
-    case "$RELEASE" in
-      jessie|stretch|buster|bullseye)
-        local e2fsprogs_version
-        # assume a more recent version if we can't identify the version via dpkg-query
-        e2fsprogs_version="$(dpkg-query --show --showformat='${Version}' e2fsprogs 2>/dev/null || echo 1.47)"
-        if [ -n "$e2fsprogs_version" ] && dpkg --compare-versions "$e2fsprogs_version" ge '1.43' ; then
-          einfo "Disabling metadata_csum_seed feature for $MKFS as $RELEASE doesn't support it."
+      bullseye|buster|stretch)
+        if [ "$mkfs_supports_metadata_csum_seed" = 1 ]; then
```
Comment (mika) [inline grml-debootstrap:1339]: Hm, isn't this logic the wrong way? What I mean:

mkfs_supports_metadata_csum_seed = 1 -> disable metadata_csum_seed

? Or am i misreading this?

Comment (zeha) [reply grml-debootstrap:1339]: this is what the code does, no?

Comment (mika) [reply grml-debootstrap:1339]: Ahhh, no I see how this is actually going, sorry4noise! :)

**Review by mika — APPROVED** [review]
Finally we're getting it under control again, thx! \o/

**Pull request merged**